jenkins/internal-ci/generic/branch/release.groovy

```diff
@@ -125,15 +125,31 @@ pipeline {
         }
         
         stage ('Sign rpm') {
-            when { expression { false } }
+            when { expression { true } }
             steps {
                 script { build_stage = env.STAGE_NAME }
                                 
                 sh label: 'Generate Key', script: '''
                     set +x
                     pushd scripts/rpm-signing
-                    cat gpgoptions >>  ~/.rpmmacros
-                    sed -i 's/passphrase/'${passphrase}'/g' genkey-batch
+                    for expiredKey in $(gpg2 --list-keys | grep -B1 "Seagate"|head -1|tr -d " ");
+                    do
+                        echo "$expiredKey"
+                        gpg --batch --yes --quiet --delete-secret-key $expiredKey >/dev/null 2>&1
+                        if [ $? -eq 0 ]; then
+                            echo "Remove old gpg secret success"
+                        else
+                            echo "Remove old gpg secret failed"
+                        fi
+                        gpg2 --batch --quiet --yes --delete-keys $expiredKey >/dev/null 2>&1
+                        if [ $? -eq 0 ]; then
+                            echo "Remove old gpg key success"
+                        else
+                            echo "Remove old gpg key failed"
```

Comment:
Should we exit if it is not able to remove old key?

Reply:
Yes. I feel we should exit out if the old key removal is failed.

Reply:
I have added exit1 command in if statement.